packages/babel-parser/src/plugins/flow.js

```diff
@@ -1730,10 +1730,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     // interfaces and enums
     parseStatement(context: ?string, topLevel?: boolean): N.Statement {
       // strict mode handling of `interface` since it's a reserved word
+      const lookahead = this.lookahead();
       if (
         this.state.strict &&
         this.match(tt.name) &&
-        this.state.value === "interface"
+        this.state.value === "interface" &&
+        (lookahead.type === tt.name || isKeyword(lookahead.value))
```

Comment:
`lookahead` is an expensive operation, could we move it after checking that `this.state.value === "interface"` is true? Otherwise we run it for _every_ statement.